src/sentry/explore/endpoints/explore_saved_query_detail.py

```diff
@@ -0,0 +1,182 @@
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NO_CONTENT,
+    RESPONSE_NOT_FOUND,
+)
+from sentry.apidocs.examples.explore_saved_query_examples import ExploreExamples
+from sentry.apidocs.parameters import ExploreSavedQueryParams, GlobalParams
+from sentry.explore.endpoints.bases import ExploreSavedQueryPermission
+from sentry.explore.endpoints.serializers import ExploreSavedQuerySerializer
+from sentry.explore.models import ExploreSavedQuery
+
+
+class ExploreSavedQueryBase(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    permission_classes = (ExploreSavedQueryPermission,)
+
+    def convert_args(self, request: Request, organization_id_or_slug, id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, organization_id_or_slug, *args, **kwargs)
+
+        try:
+            kwargs["query"] = ExploreSavedQuery.objects.get(
+                id=id,
+                organization=kwargs["organization"],
+            )
+        except ExploreSavedQuery.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return (args, kwargs)
+
+
+@extend_schema(tags=["Discover"])
+@region_silo_endpoint
+class ExploreSavedQueryDetailEndpoint(ExploreSavedQueryBase):
+    publish_status = {
+        "DELETE": ApiPublishStatus.PUBLIC,
+        "GET": ApiPublishStatus.PUBLIC,
+        "PUT": ApiPublishStatus.PUBLIC,
+    }
+
+    def has_feature(self, organization, request):
+        return features.has(
+            "organizations:performance-trace-explorer", organization, actor=request.user
+        )
+
+    @extend_schema(
+        operation_id="Retrieve an Organization's Explore Saved Query",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+            ExploreSavedQueryParams.EXPLORE_SAVED_QUERY_ID,
+        ],
+        request=None,
+        responses={
+            200: ExploreSavedQueryModelSerializer,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=ExploreExamples.EXPLORE_SAVED_QUERY_GET_RESPONSE,
+    )
+    def get(self, request: Request, organization, query) -> Response:
+        """
+        Retrieve a saved query.
+        """
+        if not self.has_feature(organization, request):
+            return self.respond(status=404)
+
+        self.check_object_permissions(request, query)
+
+        return Response(serialize(query), status=200)
+
+    @extend_schema(
+        operation_id="Edit an Organization's Explore Saved Query",
+        parameters=[GlobalParams.ORG_ID_OR_SLUG, ExploreSavedQueryParams.EXPLORE_SAVED_QUERY_ID],
+        request=ExploreSavedQuerySerializer,
+        responses={
+            200: ExploreSavedQueryModelSerializer,
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+        examples=ExploreExamples.EXPLORE_SAVED_QUERY_GET_RESPONSE,
+    )
+    def put(self, request: Request, organization, query) -> Response:
```

Comment:
We also need a `POST` to create the saved queries, right? Or were you thinking of adding that in another PR?

Reply:
POST endpoint is already merged in a different file (`explore_saved_queries.py`)!

Reply:
Oops!